### DISCUSSION TITLE
Twill API documentation

### DISCUSSION THREAD

**Discussion by pboivin** [area17/twill#921] [Ideas]
Hello Twill enthusiasts!

I've been using Laravel for a few years now. I'm just starting out with Twill and having a lot of fun!

As many of you probably are, I'm reflecting on my learning journey with Twill so far, and I'm trying to come up with ideas to add to the documentation. One thing that really helped me on the Laravel side is the [API reference](https://laravel.com/api/8.x). I'm writing this as a first step to gather general feedback on the usefulness of this type of documentation.

Personally, I find a lot of value in this "bird's eye view" of the framework. IDEs can reveal all kinds of contextually useful information while working, but I still find it challenging to really _grok_ how things are interconnected. For example, it's not always obvious where a given method comes from, when working with a class that inherits from parent classes _and_ is being composed with multiple traits...

Anyhow, I've set up a proof of concept here:
- [API docs for Twill 2.x](https://pboi20.github.io/twill-api/2.x/index.html)
- [Repository](https://github.com/pboi20/twill-api)

The approach is very much copied from the Laravel docs, so I give all credit where credit is due :)

**Comment by ifox** [comment]
@pboi20 this is fantastic, I'd accept a PR to have this running from the main project and deployed under twill.io.

**Reply by pboivin** [reply]
Will do! Thanks for the feedback :)
